usersync/syncersbuilder.go

```diff
@@ -26,21 +26,21 @@ func (e SyncerBuildError) Error() string {
 }
 
 func BuildSyncers(hostConfig *config.Configuration, bidderInfos config.BidderInfos) (map[string]Syncer, []error) {
-	// map syncer config by bidder
-	cfgByBidder := make(map[string]config.Syncer, len(bidderInfos))
+	biddersWithSyncerCfg := make(map[string]config.BidderInfo, len(bidderInfos))
 	for bidder, bidderInfo := range bidderInfos {
 		if shouldCreateSyncer(bidderInfo) {
-			cfgByBidder[bidder] = *bidderInfo.Syncer
+			biddersWithSyncerCfg[bidder] = bidderInfo
```

Comment:
I do not think you need to build bidderName to bidderInfo map again here. Should directly be able to use `bidderInfos` passed as function argument.

Reply:
I see why you need it now. To filter the biddersWithSyncerCfg to filter based on `shouldCreateSyncer` check.

Reply:
might be some missing context regarding the use of `biddersWithSyncerCfg` and `bidderInfos`.

`biddersWithSyncerCfg` is essentially a subset of `bidderInfos`, specifically holding bidder information for which a syncer config is either defined or derived. The determination of whether a syncer config should be created is made using the `shouldCreateSyncer()` function.

Additionally, if an alias has a syncer config, `biddersWithSyncerCfg` is utilized to check whether the parent also has a defined syncer config.

The reason for using `biddersWithSyncerCfg` instead of `bidderInfos` directly is to optimize the process. If we were to use `bidderInfos` directly, we would need to call `shouldCreateSyncer()` for each alias, potentially resulting in a larger number of calls to this function. Using `biddersWithSyncerCfg` allows us to avoid unnecessary calls to `shouldCreateSyncer()` and streamline the process.